src/Search/Gearman/Command/QueueCommand.php

```diff
@@ -91,9 +95,14 @@ protected function execute(InputInterface $input, OutputInterface $output)
         // Loop.
         $limit = $this->limit;
         while (!$limit()) {
```

Comment:
I think the conditions in commands can be:
```
while (!($reason = $limit())) {
    ...
}
```
where `$reason` is a string description of the reason for stopping. Then we can use it in the `$this->logger->info()` at the end of the method to say `Stopped because ...`